podcasts/MainTabBarController.swift

```diff
@@ -170,26 +170,45 @@ class MainTabBarController: UITabBarController, NavigationProtocol {
     }
 
     func navigateToPodcastInfo(_ podcastInfo: PodcastInfo) {
-        if !switchToTab(.podcasts) { return }
-
         if let navController = selectedViewController as? UINavigationController {
             navController.popToRootViewController(animated: false)
             let podcastController = PodcastViewController(podcastInfo: podcastInfo, existingImage: nil)
-            navController.pushViewController(podcastController, animated: false)
+            navController.pushViewController(podcastController, animated: true)
         }
     }
 
-    func navigateToEpisode(_ episodeUuid: String) {
+    func navigateTo(podcast searchResult: PodcastSearchResult) {
+        if let navController = selectedViewController as? UINavigationController {
+            navController.popToRootViewController(animated: false)
+            let podcastController = PodcastViewController(podcastInfo: PodcastInfo(from: searchResult), existingImage: nil)
+            navController.pushViewController(podcastController, animated: true)
+        }
+    }
+
+    func navigateToEpisode(_ episodeUuid: String, podcastUuid: String?) {
         if let navController = selectedViewController as? UINavigationController {
             navController.dismiss(animated: false, completion: nil)
 
             // I know it looks dodgy, but the episode card won't load properly if you just dismissed another view controller. Need to figure out the actual bug...but for now:
             // (before you ask, using the completion block doesn't work above, regardless of whether animated is true or false
             DispatchQueue.main.asyncAfter(deadline: .now() + 0.5.seconds) {
-                let episodeController = EpisodeDetailViewController(episodeUuid: episodeUuid, source: .homeScreenWidget)
-                episodeController.modalPresentationStyle = .formSheet
+                var episodeController: UIViewController
+
+                if EpisodeLoadingController.needsLoading(uuid: episodeUuid), let podcastUuid {
+                    episodeController = EpisodeLoadingController(episodeUuid: episodeUuid,
+                                                                 podcastUuid: podcastUuid)
```

Comment:
We can remove the var here since it's not actually used. 

```suggestion
                if EpisodeLoadingController.needsLoading(uuid: episodeUuid), let podcastUuid {
                    let episodeController = EpisodeLoadingController(episodeUuid: episodeUuid,
                                                                 podcastUuid: podcastUuid)
```